datalad/interface/tests/test_run.py

```diff
@@ -352,6 +352,7 @@ def test_rerun_just_one_commit(path):
     # in a repo" case.
     ds.repo.checkout("orph", options=["--orphan"])
     ds.repo.repo.git.reset("--hard")
+    ds.repo.config.reload()
```

Comment:
should all the smarty checks detect the necessity for config reload somehow?  how did it work before without? (worth adding a comment why needed may be?)

Reply:
It worked by accident. ;-)

Generally yes. Certain *Repo methods should know themselves to reload config. But here GitPython's reset is called, which is why the reload needs to be called in addition. What should be done is: Have a `GitRepo.reset`, that would call reload at the end and use this one here.